src/PowerShellEditorServices/Session/PowerShellVersionDetails.cs

```diff
@@ -72,5 +81,83 @@ public class PowerShellVersionDetails
             this.Edition = editionString;
             this.Architecture = architecture;
         }
+
+        #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Gets the PowerShell version details for the given runspace.
+        /// </summary>
+        /// <param name="runspace">The runspace for which version details will be gathered.</param>
+        /// <returns>A new PowerShellVersionDetails instance.</returns>
+        public static PowerShellVersionDetails GetVersionDetails(Runspace runspace)
```

Comment:
I've finally started breaking down the code in the PowerShellContext class to put it where it belongs!